storage/sealer/sched.go

```diff
@@ -381,6 +396,51 @@ func (sh *Scheduler) Info(ctx context.Context) (interface{}, error) {
 	}
 }
 
+func (sh *Scheduler) removeRequest(rmrequest *rmRequest) {
+
+	if sh.SchedQueue.Len() < 0 {
+		rmrequest.rmresE <- xerrors.New("No requests in the scheduler")
+	}
+
+	queue := sh.SchedQueue
+	for i, r := range *queue {
+		if r.SchedId == rmrequest.id {
+			queue.Remove(i)
+			rmrequest.rmresC <- struct{}{}
```

Comment:
I'd return after responding to guarantee that only one response can be sent (also avoid doing redundant work).

```suggestion
			rmrequest.rmresC <- struct{}{}
			return
```